ddtrace/internal/writer.py

```diff
@@ -489,6 +497,9 @@ def write(self, spans=None):
             )
             self._metrics_dist("buffer.dropped.traces", 1, tags=["reason:full"])
             self._metrics_dist("buffer.dropped.bytes", payload_size, tags=["reason:full"])
+        except Exception as e:
```

Comment:
Should we be sure we still `raise` at the end of this instead of just swallowing/preventing the exception?

Incrementing the error metric and a nice error log still makes sense to me 👍🏻